Adds the reverse-contractor: The Xenolife Technician.

Neon1ight opened the pull request:
Proposal thread: https://forums.baystation12.net/threads/a-modest-research-proposal-the-reverse-contractor.6973/

I am very very very new to this sort of thing. This is the most complicated thing I've done with BYOND. I still don't know how the mapping part works all that well. I'm trying my best. Inevitably, somebody will find a way to improve this. Please tell me how and I'll try.

Thank you for your help, if you did help. If it wasn't for some nice people in #coding I would've never have gotten past the mapping part, probably.

https://gyazo.com/1f8eb2ace7b2c5b80d38b257392eae6c

nearlyNonexistent commented:
no changelog

Neon1ight commented:
Going to go ahead and say this is ready for people to look at. No longer WIP. New name will be the final one.

babydoll commented on maps/torch/job/jobs.dm:
```diff
@@ -1505,6 +1505,42 @@
+	
+	total_positions = 1
+	spawn_positions = 1
+	supervisors = "the Research Director and the Commanding Officer"
```
underlings only report to their boss, not directly to the captain

Neon1ight replied:
Alright. I'll fix this.

babydoll commented on maps/torch/job/outfits.dm:
```diff
@@ -540,6 +540,24 @@ Keeping them simple for now, just spawning with basic EC uniforms, and pretty mu
+
+/decl/hierarchy/outfit/job/torch/crew/research
+	name = OUTFIT_JOB_NAME("Xenolife Technician")
+	uniform = /obj/item/clothing/under/solgov/utility/expeditionary/exploration
```
These guys aren't explorers, are not in the exploration department, and so shouldn't have an exploration uniform

Neon1ight replied:
I felt that it was the uniform that made the most sense for them. What do you propose as an alternative?

babydoll replied:
They'll need their own uniform (department stripes)

Neon1ight replied:
What color department stripes do you propose? Might need a little help setting something like that up. If I was to choose-- It'd probably be a green that matched the government research beret, but service is already green. I could try to change service to something else like gray or dark brown-- But again, might be outside my area of ability.

Neon1ight replied:
Nevermind. I think I'll try a darker, forest green, as opposed to service's pastel.

babydoll commented on maps/torch/loadout/_defines.dm:
```diff
@@ -61,7 +61,7 @@
 
 //For members of the research department and jobs that are scientific
-#define RESEARCH_ROLES list(/datum/job/rd, /datum/job/liaison, /datum/job/scientist,/datum/job/mining, /datum/job/guard, /datum/job/scientist_assistant, /datum/job/assistant, /datum/job/nt_pilot, /datum/job/senior_scientist, /datum/job/roboticist)
+#define RESEARCH_ROLES list(/datum/job/rd, /datum/job/liaison, /datum/job/scientist,/datum/job/mining, /datum/job/guard, /datum/job/scientist_assistant, /datum/job/assistant, /datum/job/nt_pilot, /datum/job/senior_scientist, /datum/job/roboticist, /datum/job/xenolife_technician)
```
You need to check to ensure that this doesn't let people circumvent uniform restrictions

Neon1ight replied:
There're already a number of jobs that're set up in a way like this (FTECH, shuttle pilot, etc) but I'll check anyway.

Neon1ight replied:
https://gyazo.com/f0b6cc7790e452811929231124e2f284 Fine on this front

babydoll commented:
Also the map bot is dead so you'll need to provide a screenshot of map changes.

Neon1ight commented:
Map change in OP. One research equipment locker (The generic, random content kind) removed in the research equipment room. Xenolife tech's locker in it's place.

Neon1ight commented:
Finished. New uniform added, with the color being NT_RED. I feel strongly that this is too similar to security, but we'll see how it plays out.

Neon1ight committed:
Adds the Xenolife Technician.

Adds the xenolife technician job, with a locker and other relevant things. See the thread for further information.
Update jobs.dm
Update solgov-accessory.dm
Update solgov-under.dm
Update solgov-suit.dm
Update solgov-accessory.dm
Update outfits.dm
Update uniforms_expedition.dm
Just for testing
Makes it NT_RED
Dron Uprising Event

Added a random event that spawns bunch of hostile maint drones in maintenance.
They are pretty weak, and are mostly intended to dissuade casual maint dipping and to provide medbay some victims.
They won't target you if you're some kind of synth, or look the part:
-Any silicon mobs
-FBPs/IPCs
-Wearing a drone on your head
-Full cardborg disguise
Fixes snow planet ore maps

They were so poor they didn't pass sanity. Now sanity is more tweakable.

Also fixes lava planet's ultra rich ore map actually being super poor.
Fixing merge conflict
Update solgov-accessory.dm
Dron Uprising Event

Added a random event that spawns bunch of hostile maint drones in maintenance.
They are pretty weak, and are mostly intended to dissuade casual maint dipping and to provide medbay some victims.
They won't target you if you're some kind of synth, or look the part:
-Any silicon mobs
-FBPs/IPCs
-Wearing a drone on your head
-Full cardborg disguise
Fixes snow planet ore maps

They were so poor they didn't pass sanity. Now sanity is more tweakable.

Also fixes lava planet's ultra rich ore map actually being super poor.

Neon1ight committed:
Quick fix

comma commented on maps/torch/items/clothing/solgov-accessory.dm:
```diff
@@ -451,6 +451,14 @@ department tags
+/obj/item/clothing/accessory/solgov/department/research
+	name = "research insignia"
+	desc = "Insignia denoting assignment to the research department. These fit Expeditionary Corps uniforms."
+	color = "#F8F8FF"
```
Gotta be lowercase. Also just leave it white (dont set color) since sabira doesnt want it to overlap with explo
